ICAFIX/scripts/RecleanClassify.sh

```diff
@@ -0,0 +1,284 @@
+opts_AddOptional '--model-to-use' 'ModelToUse' 'string' "the models to use separated by '@'" "RandomForest@MLP"
+opts_AddOptional '--vote-threshold' 'VoteThresh' 'integer' "a decision threshold for determing reclassifications, should be less than to equal to the number of models to use" ""
+
+opts_AddOptional '--matlab-run-mode' 'MatlabMode' '0, 1, or 2' "defaults to $g_matlab_default_mode
+
+0 = compiled MATLAB
+1 = interpreted MATLAB
+2 = Octave" "$g_matlab_default_mode"
+opts_ParseArguments "$@"
+
+#display the parsed/default values
+opts_ShowValues
+
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+
+case "$MatlabMode" in
+    (0)
+        if [[ "${MATLAB_COMPILER_RUNTIME:-}" == "" ]]
+        then
+            log_Err_Abort "to use compiled matlab, you must set and export the variable MATLAB_COMPILER_RUNTIME"
+        fi
+        ;;
+    (1)
+        #NOTE: figure() is required by the spectra option, and -nojvm prevents using figure()
+        matlab_interpreter=(matlab -nodisplay -nosplash)
+        ;;
+    (2)
+        matlab_interpreter=(octave-cli -q --no-window-system)
+        ;;
+    (*)
+        log_Err_Abort "unrecognized matlab mode '$MatlabMode', use 0, 1, or 2"
+        ;;
+esac
+
+UseLocalPython="TRUE"
+singularity_command=""
+if [[ "$PythonSingularity" != "" && "$PythonInterpreter" != "" ]]; then
+    log_Err_Abort "please only specify one of --python-singularity and --python-interpreter"
+fi
+
+if [[ "$PythonSingularity" != "" ]]; then
+    if [ ! -f "$PythonSingularity" ]; then
+        log_Err_Abort "the singularity container doesn't exists under python version: $PythonSingularity"
+    fi
+
+    if command -v singularity &> /dev/null; then
+        log_Msg "Singularity is installed."
+    else
+        log_Err_Abort "Singularity is not installed or not in PATH."
+    fi
+    UseLocalPython="FALSE"
+    singularity_command=(singularity exec --bind "$PythonSingularityMountPath" "$PythonSingularity" python3)
+else
+    # native env using conda
+    if [ ! -f "$PythonInterpreter" ]; then
+        PythonInterpreter="python3" # default native env
+    fi
+fi
+
+if [[ "$VoteThresh" == "" ]]; then
+    count=$(echo "$ModelToUse" | awk -F'@' '{print NF}')
+    VoteThresh=$((count - 1))
+fi
+
+IFS='@' read -a fMRINamesArray <<<"$fMRINames"
+
+tempfiles_create fMRIList_XXXXXX.txt fMRIListName
+
+# check SR or MR FIX
+if [ ! ${MRFixConcatName} = "" ] ; then
+    echo "${MRFixConcatName}" >> "$fMRIListName"
+    fMRINameToUse=${MRFixConcatName}
+else
+    fMRINameToUse=""
+    for fMRIName in "${fMRINamesArray[@]}"
+    do
+        echo "${fMRIName}" >> "$fMRIListName"
+        fMRINameToUse+=" $fMRIName"
+    done
+    # Remove the leading space
+    fMRINameToUse="${fMRINameToUse:1}"
+fi
+
+# check if FIX features are generated (csv with 181 features)
+for fMRIName in ${fMRINameToUse} ; do
```

Comment:
Consider the options in my ApplyAutoReclean suggestion thread on very similar code.